bitnami/kafka/templates/kafka-provisioning.yaml

```diff
@@ -38,30 +38,6 @@ spec:
       {{- end }}
       restartPolicy: OnFailure
       terminationGracePeriodSeconds: 0
-      initContainers:
-        - name: wait-for-available-kafka
-          image: {{ include "kafka.image" . }}
-          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
-          {{- if .Values.provisioning.containerSecurityContext.enabled }}
-          securityContext: {{- omit .Values.provisioning.containerSecurityContext "enabled" | toYaml | nindent 12 }}
-          {{- end }}
-          command:
-            - /bin/bash
-          args:
-            - -ec
-            - |
-              wait-for-port \
-                --host={{ include "common.names.fullname" . }} \
-                --state=inuse \
-                --timeout=120 \
-                {{ .Values.service.ports.client | int64 }};
-              echo "Kafka is available";
-          {{- if .Values.provisioning.resources }}
-          resources: {{- toYaml .Values.provisioning.resources | nindent 12 }}
-          {{- end }}
-        {{- if .Values.provisioning.initContainers }}
-        {{- include "common.tplvalues.render" ( dict "value" .Values.provisioning.initContainers "context" $ ) | nindent 8 }}
-        {{- end }}
```

Comment:
I'm not sure about this. Init containers are a native Kubernetes feature and one of their intended uses cases is waiting for some conditions to be met before starting the applications themselves. We do this in many of our charts.

Reply:
I realize that this is a bit of a difficult question between init containers and sidecar containers.

The problem, as indicated in the PR, is that an init container that expects fully functioning networking cannot co-exist with a sidecar that is required for functioning networking (envoy service mesh sidecar in my case). By moving the wait for port into the main container the sidecar enables it to work as intended, which I find to be a good solution without serious downsides.

The alternative would be to add an option to remove that init container, since it completely blocks the provisioning pod if used with a service mesh sidecar. But in that case I'd also like to add provisioning.preScript, so that its functionality can be supplanted by a custom script.

Reply:
(As long as I can remove the init container in some way I can use the chart, but I'd like to not need to overwrite your topic provisioning script to add a wait in the main container)

Reply:
I think that would be a great compromise.

Reply:
It has now been applied.